src/java/com/puppetlabs/ssl_utils/SSLUtils.java

```diff
@@ -438,6 +444,54 @@ public static X509CRL revokeMultiple(X509CRL crl,
         return buildCRL(crl, issuerPrivateKey, issuerPublicKey, builder);
     }
 
+    /**
+     * Given a list of certificates and a list of CRLs, validate the certificate
+     * chain, i.e. ensure that none of the certs have been revoked by checking
+     * the appropriate CRL, which must be present and currently valid.
+     * Returns nil if successful."
+     *
+     * @param certs The certificate chain to validate
+     * @param crls The CRL chain to validate
+     * @throws CertificateException
+     * @throws CertPathValidatorException
+     * @throws IOException
+     * @throws InvalidAlgorithmParameterException
+     * @throws KeyStoreException
+     * @throws NoSuchAlgorithmException
+     * @throws NoSuchProviderException
+     */
+    public static void validateCertChain(List<X509Certificate> certs,
+                                         List<X509CRL> crls)
+        throws CertificateException, CertPathValidatorException, IOException,
+               InvalidAlgorithmParameterException, KeyStoreException,
+               NoSuchAlgorithmException, NoSuchProviderException {
+        // there are bouncycastle versions of CertificateFactory and CertPath
+        // that I haven't looked into - not sure when using bouncycastle vs. java
+        // makes sense
+        final CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
+        final CertPath certPath = certFactory.generateCertPath(certs);
+        final CertPathValidator validator;
+        if (isFIPS()) {
+            validator = CertPathValidator.getInstance(PKIX_KEYMANAGER_ALGO, "BCFIPS");
+        } else {
+            validator = CertPathValidator.getInstance(CertPathValidator.getDefaultType());
+        }
+        final KeyStore truststore = certsToTrustStore(certs);
+        PKIXRevocationChecker revocationChecker = (PKIXRevocationChecker)validator.getRevocationChecker();
+        revocationChecker.setOptions(EnumSet.of(
+                PKIXRevocationChecker.Option.PREFER_CRLS,
+                PKIXRevocationChecker.Option.NO_FALLBACK));
+        PKIXBuilderParameters params =
+                new PKIXBuilderParameters(truststore, new X509CertSelector());
+        params.addCertPathChecker(revocationChecker);
+        // tests pass when this is true, false, or not included at all..
+        //params.setRevocationEnabled(true);
```

Comment:
It looks like it enabled a value that is overridden by the `addCertPathChecker` above: https://docs.oracle.com/javase/7/docs/api/java/security/cert/PKIXParameters.html#setRevocationEnabled(boolean)

Reply:
ahhh, I swear I read that, but I must've decided I got the gist before making it all the way to the end haha